quil-rs/src/instruction/mod.rs

```diff
@@ -1284,6 +1191,45 @@ impl Instruction {
         }
     }
 
+    pub fn is_valid_protoquil(&self) -> bool {
+        matches!(
+            self,
+            Instruction::Declaration(_)
+                | Instruction::Gate(_)
+                | Instruction::Pragma(_)
+                | Instruction::Reset(_)
+                | Instruction::Measurement(_)
+        )
+    }
+
+    pub fn is_valid_quilt(&self) -> bool {
```

Comment:
Yeah, sounds good to me. This is a backport from `pyQuil` that I thought might be useful. I see what you mean about them overcommitting and underdelivering. I'll deprecate or otherwise find a way to deal with it on the pyQuil side.

Reply:
These have been removed.